octobot_trading/personal_data/orders/channel/orders.py

```diff
@@ -141,6 +159,38 @@ async def handle_post_open_orders_update(
             )
             await self._complete_open_order_init(waiting_complete_init_orders, is_from_bot)
 
+    async def _complete_pending_groups_from_storage(self, pending_groups):
+        # create order groups and their associated self-managed orders if any
+        # use copy.copy(pending_groups) as new groups might be added from new orders,
+        # in this case, also create the associated new orders
+        to_complete_groups = list(pending_groups.keys())
+        completed_groups = set()
+        max_allowed_nested_chained_orders_groups = 100
+        for _ in range(max_allowed_nested_chained_orders_groups):
```

Comment:
Why do we need it?

Reply:
we do :/ it will probably almost never be useful but we need to handle the case as it is possible to loop indefinitely otherwise